### DISCUSSION TITLE
Auth hooks missing data

### DISCUSSION THREAD

**Discussion by HTMHell** [supabase/supabase#34962] [Troubleshooting]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

With [Auth Email Hooks](https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook) activated:

1. according to the docs, `email_data.email` and `email_data.new_email` should be provided, but they are not. The main problem with this, is I cannot determine the new email address. I tried looking at `email_change` from `auth.users` instead, but sometimes it's empty even after sending an email change request - so not reliable. 
2. Secure Email Changes is activated, yet I only get one webhook of `email_change`, but not `email_change_new` (I should probably receive 2 of them) - this is not critical since I can handle it from my side, but only if I can get the new email address, which I currently can't (issue 1)

## To Reproduce

1. Turn on Auth Email Hooks
2. Turn on Secure Email Changes
3. Log into an active user
4. Update the user's email using `supabase.auth.updateUser({ email: ... })`

## Expected behavior

1. Should get `email_data.email` and `email_data.new_email` in the webhook
2. (maybe?) should get `email_change_new` webhook

Here's an example of `email_data` object I recieve:

```
email_data: {
  token: '606217',
  token_hash: '7e93164b6dbb38e796b0b43b8bd72f0dad1d2fb2180a6747aac4f1ec',
  redirect_to: 'http://localhost:5173/account/settings',
  email_action_type: 'email_change',
  site_url: 'https://redacted.supabase.co/auth/v1',
  token_new: '027633',
  token_hash_new: 'e70fc2f7f107db2b8a324aaf24dc225be4e35a2bcd9ae77875b25987'
}
```

**Comment by dnywh** [comment]
Confirming the same issue on my end. This is pretty important as this `email_data.new_email` (or `email_change_new`) is required to send that email verification link _to the new email address_.

Without that information, I'm having to fall back to emailing the old email to confirm the new one.

The raw JSON for the user shows this `email_change`. I'm just not able to access it via the web hook.

```json
{
  "email": "test@peels.app",
  "email_change": "test1@peels.app",
  },
  "raw_user_meta_data": {
    "email": "test@peels.app",
  },
```

**Comment by dnywh** [comment]
I've found a solution by using `user.new_email`. I found this accidentally by poking around the user JSON [similar issue](https://github.com/supabase/auth/issues/1744#issue-2490654011). 

I assume `new_email` has since been swapped out for `email_change` everywhere else except for the web hook, which still looks for the old `new_email`.

**Comment by Hallidayo** [comment]
Hi everyone, I'm going to move the issue over to discussions/enhancements. Thank you @dnywh for your help on this one 👍
